gigascience/gigadb-website
# A11y - Improve accessibility of login page

# Pull request for issue: #1384

This is a pull request for the following functionalities:

A handful of accessibility improvements in the login page: http://gigadb.gigasciencejournal.com:9170/site/login

![image](https://github.com/gigascience/gigadb-website/assets/143437854/ef41a748-237d-4598-a3f5-3b846ccaa9a4)


* Enhance input focus state to make it more visible
* Moved links to other pages outside the form and below the login button for a simpler UX
* Colored errors with red
* Wrapped errors in a role="alert" so that they are announced to screen readers
* Added required attribute to inputs to leverage client side validation
* Hide asterisk from screen readers as the input is announced as required

## How to test?

* Navigate the login form with keyboard
* Try to submit the form with empty inputs to see client side validation (the default browser-provided validation for required inputs)
* Submit form with invalid credentials to see error

Note: the checkbox focus state is handled in another PR

## How have functionalities been implemented?

See functionalities

## Any issues with implementation?

None

## Any changes to automated tests?

None

## Any changes to documentation?

None

## Any technical debt repayment?

None

## Any improvements to CI/CD pipeline?

None
